### ISSUE TITLE
Yolov8 segmentation

### ISSUE THREAD

**Issue by HSVDO** [vladiH/flutter_vision#19]
Hi,

first of all I'd like to thank you for your work. I tried various ways (pytorch mobile via torchscript and tflite_flutter library) to get Yolov8 running in flutter but your library was the only one which worked for me with the default Yolov8 model. 

So currently I am trying to get the Yolov8 segmentation running (using official yolov8-seg models). I figured out that the output shape of yolov8-seg differs from detection task's output. 

Yolov8 CLI export to tflite prints: `PyTorch: starting from yolov8n-seg.pt with input shape (1, 3, 640, 640) BCHW and output shape(s) ((1, 116, 8400), (1, 32, 160, 160)) (6.7 MB)`

I added a logging statement to your load model method to see the shape in tflite files: `Shapes retrieved: [Input tensor #0 has shape: [1, 640, 640, 3], Output tensor #0 has shape: [1, 116, 8400], Output tensor #1 has shape: [1, 160, 160, 32]].`

This gives following output for default yolov8 model:
`PyTorch: starting from     yolov8n.pt with input shape (1, 3, 640, 640) BCHW and output shape(s) (1, 84, 8400) (6.2 MB)`
`Shapes retrieved: [Input tensor #0 has shape: [1, 224, 224, 3], Output tensor #0 has shape: [1, 84, 1029]].`

As I am pretty new to machine learning I am a little bit confused why the loaded model has a lower image size and different output shape. Can you give me a hint on this?

So I went on and tried to modify the output output array in `Yolo.java`. Now the detection task doesn't give a message like `java.lang.IllegalArgumentException: Cannot copy to a TensorFlowLite tensor (inputs_0) with 4915200 bytes from a Java Buffer with 602112 bytes.` anymore but causes always an OutOfMemory error. 

I'd really appreciate your support to get segmentation running!
Thanks in advance :)

**Comment by rahulmanuwas** [comment]
@HSVDO any solutions so far? We are also trying to run a custom segmentation model based on yolo v8 on flutter.

**Comment by HSVDO** [comment]
Hey there, I was on vacation the last few days,  currently I am on the train home. I hope I can invest some time this week. But as I mentioned above, I am pretty new to machine learning and it is kind of "try and error"...

**Comment by HSVDO** [comment]
@rahulmanuwas the error `java.lang.IllegalArgumentException: Cannot copy to a TensorFlowLite tensor (inputs_0) with 4915200 bytes from a Java Buffer with 602112 bytes.` seems to appear when loading a second model after a detection model was loaded initially. So I changed my app to show the segmentation screen first instead of detection screen. Then I get some results from Yolo but they are still a little bit confusing to me as the labels create an ArrayIndexOutOfBoundsException and I have to do further investigations...

The mentioned OOM was triggered by my try to instantiate the output array with too many too big dimensions.

Have you had any success with the segmentation yet?

**Comment by HSVDO** [comment]
@rahulmanuwas just to give you a brief update: I guess I made some important progress. So far I am able to retrieve the boxes & classification as well as the second output using the segmentation model yolov8n-seg. Next step is to use the second output. After that I will do some refactoring and then I will push it to my fork.

The comments in https://github.com/ultralytics/ultralytics/issues/2953 explain the output of Yolov8 segmentation very well.

**Comment by HSVDO** [comment]
@rahulmanuwas I managed to get things nearly done, I guess. The segmentation output is now returned to dart. However I'm not sure yet how to handle the prototype masks returned by Yolo.

You can have a look at: https://github.com/HSVDO/flutter_vision

**Comment by vladiH** [comment]
New release flutter_vision 1.1.3
